### PR TITLE
fixes #13113 - Add roles name in the title of the page

### PR DESCRIPTION
**Pull request by stuxboulot**
Display the role name in title of the page in case of this is present.
If the user use the search bar for filter the role, the string filter is only use in the title (in this case, th rolesname is in the searchbar)

### CODE REVIEW

File: app/views/filters/index.html.erb [hunk 1]
```diff
@@ -1,4 +1,8 @@
-<% title _("Filters") %>
+<% if @role
+     title _("Filters for role") + " " + @role.to_s
```
Comment (domcleal) [inline app/views/filters/index.html.erb]: To translate this correctly, it should be: `title _("Filters for role %s") % @role.to_s` which will allow translators to move the %s around as they like (http://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers has more tips).

**Comment by domcleal** [comment]
[test] will run our tests for you.  I sent a GitHub invitation to our organisation if you would like to join, then it will test automatically.  You will need to change visibility from Private to Public on https://github.com/orgs/theforeman/people after accepting the invite.

Could you also add a test for this?  I'd suggest editing `test/functional/filters_controller_test.rb` which relates to this page.  There's a test for "should get index with role_id and create scoped search on role.name" already, so you could add this to the bottom of that test:

```
assert_match "Filters for role Manager", @response.body
```

**Comment by stuxboulot** [comment]
Thanks for all information. If you have other best pactice, or technical documentation about foreman, can you send me. I'am not an expert in ruby, i just discovert it for some days and not rails...
for exemple, have you some document that explore the authentication rules?

Thanks

**Comment by domcleal** [comment]
There's not a lot, but what we have is on the [development resources](http://projects.theforeman.org/projects/foreman/wiki/Development_Resources) page (see "Granular permission system" in particular) and the [handbook](http://theforeman.org/handbook.html) web page.

**Comment by domcleal** [comment]
[test]

**Comment by stuxboulot** [comment]
i discover an error in the last commit. Now i discover how locallytest case.
I will adapt my workflow for send only tested commit.
Sorry

**Comment by domcleal** [comment]
No worries.  [test]

**Comment by stuxboulot** [comment]
For information, the new therms Rules for role %s will be available on transifex automatically when the pullrequest is accepted ?

**Comment by domcleal** [comment]
It would take a little while - this would go into the develop/1.11 branch, and Transifex currently still has the 1.10 branch for translation.  After 1.10.1 or .2 is released I'll probably switch it over to 1.11 and it'll appear then.

**Comment by stuxboulot** [comment]
another information.
If y want to add a "big" change or a big fonctionnallity, what is the workflow for accpet this?
I need to develop and propose a pull request, or if a create the feature on readmine, a board validate the need, before i develop that?

for exemple, on the host detail page, i like to add a tab with the puppet status, this tab report the list of the affected class of the node with is status (ok, pending, failed) i think this work is conscequent by very interessent for who use foreman like an ENC puppet (it's my case actually, don't use provitionning functionnallity...)

perapse i need to post this on formen mailing list?

**Comment by domcleal** [comment]
Usually just file a Redmine ticket, develop it and send a pull request.  The more PRs and tickets you can split it logically into, the easier it usually is.  If you are unsure about whether the feature would be accepted then discussing it on the foreman-dev list or a Redmine ticket may be a good idea, but it sounds like something that probably belongs in Foreman with the existing Puppet and reports functionality.

**Comment by domcleal** [comment]
Thanks for the patch @stuxboulot, merged as e50d587.

**Pull request closed without merging**